modules/zip/src/parse-zip/zip64-info-generation.ts

```diff
@@ -0,0 +1,96 @@
+export const signature = new Uint8Array([0x01, 0x00]);
+
+/** info that can be placed into zip64 field */
+type Zip64Options = {
+  /** Original uncompressed file size and Size of compressed data */
+  size?: number;
+  /** Offset of local header record */
+  offset?: number;
+};
+
+/**
+ * creates zip64 extra field
+ * @param options info that can be placed into zip64 field
+ * @returns buffer with field
+ */
+export const createZip64Info = (options: Zip64Options): ArrayBuffer => {
+  const optionsToUse = {
+    ...options,
+    zip64Length: (options.offset ? 1 : 0) * 8 + (options.size ? 1 : 0) * 16
+  };
+
+  let result = new ArrayBuffer(0);
```

Comment:
Repeated concatenation is expensive (you end up copying the same data over and over again). Better to save an array of chunks and concatenate at the end.
```suggestion
  const chunks: ArrayBuffer[] = [];
  ...
  return concatenateArrayBuffers(chunks);
```